src/cli/tools/gleam.ts

```diff
@@ -0,0 +1,80 @@
+import fs from 'node:fs/promises';
+import { join } from 'node:path';
+import { execa } from 'execa';
+import { inject, injectable } from 'inversify';
+import { InstallToolBaseService } from '../install-tool/install-tool-base.service';
+import {
+  CompressionService,
+  EnvService,
+  HttpService,
+  PathService,
+} from '../services';
+
+@injectable()
+export class InstallGleamService extends InstallToolBaseService {
+  readonly name = 'gleam';
+
+  private get ghArch(): string {
+    switch (this.envSvc.arch) {
+      case 'arm64':
+        return 'aarch64';
+      case 'amd64':
+        return 'x86_64';
+    }
+  }
+
+  constructor(
+    @inject(EnvService) envSvc: EnvService,
+    @inject(PathService) pathSvc: PathService,
+    @inject(HttpService) private http: HttpService,
+    @inject(CompressionService) private compress: CompressionService,
+  ) {
+    super(pathSvc, envSvc);
+  }
+
+  override async install(version: string): Promise<void> {
+    /**
+     * @example
+     * @see {@href https://github.com/gleam-lang/gleam/releases/tag/v0.34.1}
+     */
+    const baseUrl = `https://github.com/gleam-lang/gleam/releases/download/v${version}/`;
+
+    const filename = `gleam-v${version}-${this.ghArch}-unknown-linux-musl.tar.gz`;
+    const url = `${baseUrl}${filename}`;
+
+    const checksumFile = await this.http.download({
+      url: `${url}.sha512`,
+    });
```

Comment:
this will fail on gleam `<0.19-rc1`, as there are no checksum files.

https://github.com/gleam-lang/gleam/releases?page=8

i think we can simply override the `validate` method and ensure a version `>=0.19.0-rc1`

Reply:
sure, np. [bd6c9ca](https://github.com/containerbase/base/pull/2018/commits/bd6c9ca419c366b653e9a45678e271a58a1d25b5)